File: sorting/C_Cpp/sort_colors_bucket.c

void sortColors(int* nums, int numsSize) {
    int counts[3] = {0, 0, 0};
    for (int i = 0; i < numsSize; i++) {
        counts[nums[i]]++;
    }
    int c = 0;
    for (int i=0; i<3; i++) {
        for (int j=0; j < counts[i]; j++) {
            nums[c] = i;
            c++;
        }
    }
}

File: sorting/Python/sort_colors_bucket.py
#sort in place
class Solution:
    def sortColors(self, nums: list[int]) -> None:
        counts = [0,0,0]
        for n in nums:
            counts[n] += 1
        c = 0
        for i in range(len(counts)):
            for j in range(counts[i]):
                nums[c] = i
                c+=1
